lib/Source.ml

```diff
@@ -302,31 +309,6 @@ let char_literal t (l : Location.t) =
         (Literal_lexer.char (string_at_loc t loc))
   | _ -> impossible "Pconst_char is only produced by char literals"
 
-let begins_line ?(ignore_spaces = true) t (l : Location.t) =
-  let rec begins_line_ cnum =
-    cnum = 0
-    ||
-    let cnum = cnum - 1 in
-    match t.text.[cnum] with
-    | '\n' | '\r' -> true
-    | c when Char.is_whitespace c && ignore_spaces -> begins_line_ cnum
-    | _ -> false
-  in
-  begins_line_ l.loc_start.pos_cnum
-
-let ends_line t (l : Location.t) =
-  let len = String.length t.text in
-  let rec ends_line_ cnum =
-    if cnum >= len then true
-    else
-      match t.text.[cnum] with
-      | '\n' | '\r' -> true
-      | c when Char.is_whitespace c -> ends_line_ (cnum + 1)
-      | _ -> false
-  in
-  if Location.compare Location.none l = 0 then false
-  else ends_line_ l.loc_end.pos_cnum
```

Comment:
I understand the benefit to distinguish between string or token based functions, but this produces some noise in the diff, maybe they can be rewritten with a token analysis later anyway.

Reply:
you can review commit by commit

Reply:
Yes but these commits will be squashed when merging

Reply:
done